src/linear_algebra/finsupp.lean

```diff
@@ -786,6 +786,24 @@ end finsupp
 variables {R : Type*} {M : Type*} {N : Type*}
 variables [semiring R] [add_comm_monoid M] [module R M] [add_comm_monoid N] [module R N]
 
+section
+variables (R)
+/--
+Pick some representation of `x : span R w` as a linear combination in `w`,
+using the axiom of choice.
+-/
+def span.repr (w : set M) :
+  span R w → (w →₀ R) :=
+λ x, ((finsupp.mem_span_iff_total _ _ _).mp x.2).some
```

Comment:
Why not bind `x` in the parameter list?
```suggestion
def span.repr (w : set M) (x : span R w) :
  w →₀ R :=
((finsupp.mem_span_iff_total _ _ _).mp x.2).some
```

Reply:
I was also wondering whether we should unbundle `x` into `(x : M) (hx : x ∈ span R w)`, but on reflection keeping it bundled makes sense to prevent "creating data from props".